extensions/data-plane/data-plane-azure-storage/src/test/java/org/eclipse/edc/connector/dataplane/azure/storage/DestinationBlobNameTest.java

```diff
@@ -0,0 +1,94 @@
+    @CsvSource(value = {
+            "partName, null, folderName, 1, folderName/partName",
+            "partName, null, folderName, 2, folderName/partName",
+            "partName, ''  , folderName, 1,  folderName/partName",
+            "partName, ''  , folderName, 2,  folderName/partName",
+            "partName, null, folderName/, 1, folderName/partName",
+            "partName, null, folderName/, 2, folderName/partName",
+            "partName, ''  , folderName/, 1, folderName/partName",
+            "partName, ''  , folderName/, 2, folderName/partName",
+    }, nullValues = { "null" })
+    void shouldAppendFolderNameWithPartNameUsingSlash_whenBlobNameIsNullOrEmpty(String partName, String blobName, String folderName, int partSize, String expected) {
+        destinationBlobName.setBlobName(blobName);
+        destinationBlobName.setFolderName(folderName);
+        assertEquals(expected, destinationBlobName.resolve(partName, partSize));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, '',   '', 1, partName",
+            "partName, '',   '', 2, partName",
+            "partName, null,   null, 1, partName",
+            "partName, null,   null, 2, partName",
+            "partName, null,   '', 1, partName",
+            "partName, null,   '', 2, partName",
+            "partName, '',   null, 1, partName",
+            "partName, '',   null, 2, partName",
+
+    }, nullValues = { "null" })
+    void shouldUsePartName_whenBothBlobNameAndFolderNameIsNullOrEmpty(String partName, String blobName, String folderName, int partSize, String expected) {
+        destinationBlobName.setBlobName(blobName);
+        destinationBlobName.setFolderName(folderName);
+        assertEquals(expected, destinationBlobName.resolve(partName, partSize));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName, folderName, 1, folderName/blobName",
+            "partName, blobName,  folderName/, 1,  folderName/blobName",
+    }, nullValues = { "null" })
+    void shouldAppendFolderNameWithBlobNameUsingSlash_whenPartSizeEquals1(String partName, String blobName, String folderName, int partSize, String expected) {
+        destinationBlobName.setBlobName(blobName);
+        destinationBlobName.setFolderName(folderName);
+        assertEquals(expected, destinationBlobName.resolve(partName, partSize));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+            "partName, blobName, folderName, 2, folderName/partName",
+            "partName, blobName,  folderName/, 2,  folderName/partName",
+    }, nullValues = { "null" })
+    void shouldAppendFolderNameWithPartNameUsingSlash_whenPartSizeIsGraterThan1(String partName, String blobName, String folderName, int partSize, String expected) {
+        destinationBlobName.setBlobName(blobName);
+        destinationBlobName.setFolderName(folderName);
+        assertEquals(expected, destinationBlobName.resolve(partName, partSize));
+    }
```

Comment:
I think the case with the folder name not ending with a slash could be tested in a dedicated test and have all the other tests not to considerate that case, otherwise we're asserting the same behavior multiple time (not harmful at all but definitely not needed)